### PR TITLE
improve description of Extensions

### PR DESCRIPTION
**Pull request by kmaehashi**
I separated each Extensions into category and added description.

### CODE REVIEW

File: docs/source/reference/training.rst [hunk 1]
```diff
@@ -35,33 +35,93 @@ Updater
...
 
-Extension
----------
+Extensions
+----------
+
+An extension is a callable object that can perform arbitrary actions during the training loop.
+Extensions can be registered to :class:`Trainer` by using :func:`Trainer.extend` method, and they are invoked when the :ref:`Trigger <triggers>` condition is satisfied.
+
+In addition to the built-in extensions listed below, you can define your own extension using :class:`Extension` or :meth:`make_extension` decorator.
```
Comment (hvy) [inline docs/source/reference/training.rst]: How about this small change `... own extension by implementing Extension or using the make_extension() decorator`?

File: docs/source/reference/training.rst [hunk 2]
```diff
@@ -35,33 +35,93 @@ Updater
...
+   :toctree: generated/
+   :nosignatures:
+
+   chainer.training.extensions.ExponentialShift
+   chainer.training.extensions.LinearShift
+
+Reporting
+~~~~~~~~~
+
+These extensions provide features to perform reporting of metrics and various statistics to console or files.
```
Comment (hvy) [inline docs/source/reference/training.rst]: How about adding a `the`, `... to the console ...`.

**Comment by hvy** [comment]
LGTM except two small comments.

**Comment by kmaehashi** [comment]
Thanks for review! Fixed accordingly.

**Comment by hvy** [comment]
Thanks for the quick fixes. LGTM!

**Pull request merged**